File: python/kth_largest_element_in_an_array.py
# https://leetcode.com/problems/kth-largest-element-in-an-array/

import heapq

class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        # heapify(nums) # 리스트를 최소 heap으로 바꿈. (인자로 넣은 리스트 자체를 직접 변환)
        
        # heap 정렬을 이용하여 nums 배열에서 k개의 가장 큰 원소를 추출하여 리스트로 return
        return nlargest(k, nums)[-1]

File: java/leetcode/KthLargestElementInAnArray.java
package leetcode;

import java.util.PriorityQueue;
import java.util.Queue;

//https://leetcode.com/problems/kth-largest-element-in-an-array/description/

class Solution {
    public int findKthLargest(int[] nums, int k) {
        Queue<Integer> pq = new PriorityQueue<>((n1, n2) -> n2 - n1);
        for (int num : nums)
            pq.add(num);

        for (int i = 0; i < k - 1; i++)
            pq.poll();

        return pq.poll();
    }
}